backup/src/main/java/io/camunda/zeebe/backup/management/InProgressBackupImpl.java

```diff
@@ -157,7 +168,26 @@ public ActorFuture<Void> findSnapshotFiles() {
 
   @Override
   public ActorFuture<Void> findSegmentFiles() {
-    return concurrencyControl.createCompletedFuture();
+    final ActorFuture<Void> filesCollected = concurrencyControl.createFuture();
+    try (final var stream = Files.list(segmentsDirectory)) {
+      final var fileSet =
+          stream
+              .filter(isSegmentsFile)
+              .collect(Collectors.toMap(path -> path.toFile().getName(), path -> path));
```

Comment:
:wrench: we could also do it like this:

```java
.collect(
    Collectors.toMap(
        path -> segmentsDirectory.relativize(path).toString(), path -> path));
```

this would skip the `.toFile()` and also support (hypothetical) nested segment file layout.